Source/objects.cpp

```diff
@@ -3318,15 +3318,13 @@ void OperateL3Door(int pnum, int i, bool sendflag)
 
 void OperatePedistal(int pnum, int i)
 {
-	BYTE *mem;
-	int iv;
-
 	if (numitems >= MAXITEMS) {
 		return;
 	}
 
-	if (object[i]._oVar6 != 3 && PlrHasItem(pnum, IDI_BLDSTONE, &iv) != NULL) {
-		RemoveInvItem(pnum, iv);
+	int iv = plr[pnum].GetItemIndex(IDI_BLDSTONE);

```

Comment:
Same as the other one. This does not short-circuit and will retrieve the item when not needed.